packages/crayons-core/src/components/file-uploader/progress/file-uploader-progress.tsx

```diff
@@ -0,0 +1,107 @@
+import { Component, Prop, Event, EventEmitter, h } from '@stencil/core';
+
+@Component({
+  tag: 'fw-file-uploader-progress',
+  styleUrl: 'file-uploader-progress.scss',
+  shadow: true,
+})
+export class FileUploaderProgress {
+  /**
+   * file Id
+   */
+  @Prop() fileId!: number;
+
+  /**
+   * file name
+   */
+  @Prop() fileName = '';
+
+  /**
+   * file upload progress
+   */
+  @Prop() progress = 0;
+
+  /**
+   * error text for the file upload
+   */
+  @Prop() error = '';
+
+  /**
+   * retryUpload event to emit in case of a retry
+   */
+  @Event() retryUpload: EventEmitter;
+
+  /**
+   * private
+   * @param fileId
+   */
+  retryFileUpload(fileId) {
+    this.retryUpload.emit({ fileId });
```

Comment:
Who will listen to this?

Reply:
If in case a file upload files. We have enabled a retry option for individual files. The parent element, file-uploader listens to this event and does the server call again.
<img width="346" alt="Screenshot 2022-03-08 at 8 39 20 AM" src="https://user-images.githubusercontent.com/61269786/157158889-479ddbaf-722f-4eef-b508-9e5bc827134e.png">